CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contributing
+
+Thank you for your interest in contributing to Celestia-Node! 
+
+All work on the code base should be motivated by [our Github
+issues](https://github.com/celestiaorg/celestia-node/issues). If you
+would like to work on an issue which already exists, please indicate so
+by leaving a comment.
+
+All new contributions should start with a [Github
+issue](https://github.com/celestiaorg/celestia-node/issues/new/choose). The
+issue helps capture the problem you're trying to solve and allows for
+early feedback. Once the issue is created the process can proceed in different
+directions depending on how well-defined the problem and potential
+solution are. If the change is simple and well understood, maintainers
+will indicate their support with a heartfelt emoji.
+
+When the problem is well understood but the solution leads to large structural
+changes to the code base, these changes should be proposed in the form of an
+[Architectural Decision Record (ADR)](./docs/adr/). The ADR will help
+build consensus on an overall strategy to ensure the code base maintains
+coherence in the larger context. If you are not comfortable with writing an
+ADR, you can open a less-formal issue and the maintainers will help you turn it
+into an ADR.
+
+> How to pick a number for the ADR?
+
+Find the largest existing ADR number and bump it by 1.
+
+Each stage of the process is aimed at creating feedback cycles which align contributors and maintainers to make sure:
+
+- Contributors don’t waste their time implementing/proposing features which won’t land in master.
+- Maintainers have the necessary context in order to support and review contributions.
+
+## PR Naming
+
+PRs should be titled as following: 
+```txt
+pkg: Concise title of PR
+```
+
+**Example:**
+
+```txt
+service/header: Remove race in core_listener
+```
+
+## Changelog
+
+Every *notable* fix, improvement, feature, or breaking change should be made in a
+pull-request that includes an update to the `CHANGELOG_PENDING.md` file.
+
+Changelog entries should be formatted as follows:
+
+```md
+- [module/pkg: Some description about the change #xxx](link to PR) [@contributor](link to contributer github) 
+```
+
+Here, `module` is the part of the code that changed (typically a
+top-level Go package), `xxx` is the pull-request number, and `contributor`
+is the author/s of the change.
+
+## Branching Model and Release
+
+The main development branch is main.
+
+Every release is maintained in a release branch named `vX.Y.Z`.
```

Comment:
This sounds like the release branch is named `vX.Y.Z` while in fact they are (or should be) called `vX.Y` as mentioned below as well. Or am I misunderstanding sth here?

Reply:
<img width="149" alt="Screenshot 2022-01-26 at 11 33 43" src="https://user-images.githubusercontent.com/41963722/151147817-dc9460ad-b036-4b6f-a74d-b4638e2e2caf.png">

Reply:
That is a tag. Not a (necessarily) a branch.

Reply:
I think @Wondertan in the past created a branch as well which was called v0.1.1. But it seems he has deleted that branch as well now? 

Either way: for releases we should create a branch with the name `vX.Y` and on that we tag the releases `vX.Y.0`, `vX.Y.1` and so on.

Reply:
Agree with @liamsi